python/ray/serve/_private/deployment_state.py

```diff
@@ -1079,6 +1108,25 @@ def get_running_replica_infos(self) -> List[RunningReplicaInfo]:
             for replica in self._replicas.get([ReplicaState.RUNNING])
         ]
 
+    def get_replica_details(self) -> List[ReplicaDetails]:
+        details = []
+        for state in ReplicaState:
+            for replica in self._replicas.get([state]):
+                replica_details = ReplicaDetails(
+                    replica_id=replica.replica_tag,
+                    state=state,
+                    pid=replica.actor_pid,
+                    actor_name=replica._actor._actor_name,
+                    actor_id=replica.actor_id,
+                    node_id=replica.actor_node_id,
+                    node_ip=replica.actor_node_ip,
+                    start_time_s=replica._start_time,
+                )
```

Comment:
should we just store `ReplicaDetails` in the `ReplicaState` to avoid having more duplicate structs?

Reply:
Currently `ReplicaState` is a string enum representing the status of a replica - I'm not sure if we can change `ReplicaState` to store more values. Are you referring to another class?

Reply:
oh i'm sorry I meant whatever the type this `replica` is that we're translating all of the fields from

Reply:
Ah gotcha. The fields are being pulled from [`DeploymentReplica`](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L655). 
Do you mean to replace `ReplicaDetails` with `DeploymentReplica`? It seems like we could convert this into a dataclass so it can be properly validated when placed into `DeploymentDetails`, but it also has references to the actor and actor handle, and it has methods for `start`, `recover`, etc which makes it feel not appropriate. WDYT?

Reply:
I'm suggesting that we embed `ReplicaDetails` within `DeploymentReplica` instead of having the fields that we're pulling the info from here. Then we can just pull out the `ReplicaDetails` struct directly.

Reply:
Synced offline, this is a nice improvement but might need a workaround for `ReplicaState` not being tracked in `DeploymentReplica`. Going with [Sihan's suggestion](https://github.com/ray-project/ray/pull/33292#discussion_r1137533232) for a first-step improvement for now.